bundle.proto

```diff
@@ -102,10 +104,88 @@ message BundleResult {
     // Rejected by the block-engine.
     Rejected rejected = 3;
     // Reached finalized commitment level.
-    Finalized finalized = 4;
+    FinalizedLegacy finalized = 4;
     // Reached a processed commitment level.
-    Processed processed = 5;
+    ProcessedLegacy processed = 5;
     // Was accepted and forwarded by the block-engine but never landed on-chain.
     Dropped dropped = 6;
   }
 }
+
+/** BundleStatus types */
+
+enum CommitmentLevel {
+  Processed = 0;
+  Confirmed = 1;
+  Finalized = 2;
+}
+message LandedInBlock {
+  string validator_identity = 1;
+  uint64 slot = 2;
+  uint64 bundle_start_index = 3;
+  uint64 bundle_end_index = 4;
+  CommitmentLevel commitment_level = 5;
+}
+
+// Indicates the bundle won an auction and was forwarded to validators.
+message ForwardedToValidator {
+  // Point in time at which the bundle was forwarded.
+  uint64 slot_forwarded_at = 1;
+}
+
+// Bundle expired and pruned from mempool.
+message ExpiredAndPruned {
+  // Bundle expirations are determined by the transaction with the oldest blockhash.
+  string expired_signature = 1;
+
+  // The slot this blockhash was checked at.
+  uint64 slot_checked_at = 2;
+
+  // The expiration slot of the blockhash.
+  uint64 slot_expired_at = 3;
+}
+
+// Bundle contains transactions that have already processed on-chain.
+message ContainsAlreadyProcessedSignatures {
```

Comment:
this doesnt seem to be used?

Reply:
Maybe can fold this into the simulation failure error message. Looking